#!/bin/sh -e
##
## 10_move_asound_state_to_var.dpatch by David B. Harris
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Move asound.state location to /var/lib/alsa/

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

@DPATCH@
diff -urNad alsa-utils-1.0.9a/alsaconf/alsaconf.in /tmp/foo/alsa-utils-1.0.9a/alsaconf/alsaconf.in
--- alsa-utils-1.0.9a/alsaconf/alsaconf.in	2005-03-24 09:24:52.000000000 +0100
+++ /tmp/foo/alsa-utils-1.0.9a/alsaconf/alsaconf.in	2005-06-03 16:01:48.000000000 +0200
@@ -705,8 +705,8 @@
 	  aplay -N $TESTSOUND
       fi
     fi
-    if [ ! -r /etc/asound.state ]; then
-	xecho "Saving the mixer setup used for this in /etc/asound.state."
+    if [ ! -r /var/lib/alsa/asound.state ]; then
+	xecho "Saving the mixer setup used for this in /var/lib/alsa/asound.state."
 	$sbindir/alsactl store
     fi
     clear
diff -urNad alsa-utils-1.0.9a/alsactl/alsactl.c /tmp/foo/alsa-utils-1.0.9a/alsactl/alsactl.c
--- alsa-utils-1.0.9a/alsactl/alsactl.c	2005-06-03 16:02:46.000000000 +0200
+++ /tmp/foo/alsa-utils-1.0.9a/alsactl/alsactl.c	2005-06-03 16:02:14.000000000 +0200
@@ -30,7 +30,7 @@
 #include <alsa/asoundlib.h>
 #include "alsactl.h"
 
-#define SYS_ASOUNDRC "/etc/asound.state"
+#define SYS_ASOUNDRC "/var/lib/alsa/asound.state"
 #define SYS_ASOUNDNAMES "/etc/asound.names"
 
 int debugflag = 0;
diff -urNad alsa-utils-1.0.9a/alsactl/alsactl.1 /tmp/foo/alsa-utils-1.0.9a/alsactl/alsactl.1
--- alsa-utils-1.0.9a/alsactl/alsactl.1	2005-05-10 12:55:24.000000000 +0200
+++ /tmp/foo/alsa-utils-1.0.9a/alsactl/alsactl.1	2005-06-03 16:03:14.000000000 +0200
@@ -39,7 +39,7 @@
 
 .TP
 \fI\-f, \-\-file\fP
-Select the configuration file to use. The default is /etc/asound.state or
+Select the configuration file to use. The default is /var/lib/alsa/asound.state or
 /etc/asound.names (for the \fInames\fP command).
 
 .TP
@@ -56,7 +56,7 @@
 Print alsactl version number.
 
 .SH FILES
-\fI/etc/asound.state\fP (or whatever file you specify with the
+\fI/var/lib/alsa/asound.state\fP (or whatever file you specify with the
 \fB\-f\fP flag) is used to store current settings for your
 soundcards. The settings include all the usual soundcard mixer
 settings.  More importantly, alsactl is
